### PR TITLE
Add openAPIV3Schema validation to Helm CRD

### PR DESCRIPTION
**Pull request by stefanprodan**
- make git path and values required fields
- set FluxHelmRelease short name to fhr
- prep for multi-version CRD
- cleanup old chart archives

### CODE REVIEW

File: deploy-helm/flux-helm-release-crd.yaml [hunk 1]
```diff
@@ -9,5 +9,27 @@ spec:
...
+  versions:
+    - name: v1alpha2
+      served: true
+      storage: true
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - releaseName
```
Comment (squaremo) [inline deploy-helm/flux-helm-release-crd.yaml]: This isn't required -- it's only used if you want to override the release name that would be generated from the namespace and name.

**Comment by stefanprodan** [comment]
@squaremo the CRD validation was promoted to beta in Kubernetes 1.9

**Comment by stefanprodan** [comment]
I will squash the commits on merge, I made a mess

**Comment by squaremo** [comment]
> I will squash the commits on merge, I made a mess

Squash _before_ merging, please.

**Comment by squaremo** [comment]
You squashed _on_ merging.

**Pull request merged**